common/src/main/java/tachyon/network/protocol/RPCMessageEncoder.java

```diff
@@ -42,9 +42,9 @@ protected void encode(ChannelHandlerContext ctx, RPCMessage in, List<Object> out
     long bodyBytes = 0;
     DataBuffer payload = null;
 
-    if (in.hasPayload()) {
-      payload = in.getPayloadDataBuffer();
-      bodyBytes = payload.getLength();
+    if (in instanceof PayloadMessage) {
+      payload = ((PayloadMessage)in).getPayloadDataBuffer();
+      bodyBytes = (payload == null) ? 0 : payload.getLength();
```

Comment:
yeah that was unexpected.. i am not sure that it happens in real usage, but most of the integration tests use  a null value for the payload. So when I did not have the check, a bunch of tests failed.